sdk/src/crypto/ocsp/fetch.rs

```diff
@@ -29,7 +33,99 @@ use crate::crypto::base64;
+    url: url::Url,
+}
+
+// Common function to extract AIA responders from a certificate
+fn extract_aia_responders(cert: &x509_parser::certificate::X509Certificate) -> Option<Vec<String>> {
+    let em = cert.extensions_map().ok()?;
+
+    let aia_extension = em.get(&AUTHORITY_INFO_ACCESS_OID)?;
+
+    let ParsedExtension::AuthorityInfoAccess(aia) = aia_extension.parsed_extension() else {
+        return None;
+    };
+
+    let mut output = Vec::new();
+
+    for ad in &aia.accessdescs {
+        if let x509_parser::extensions::GeneralName::URI(uri) = ad.access_location {
+            if ad.access_method == AD_OCSP_OID {
+                output.push(uri.to_string())
+            }
+        }
+    }
+    Some(output)
+}
+
+// Common function to build OCSP request data
+fn build_ocsp_request(certs: &[Vec<u8>], responder_url: &str) -> Option<OcspRequestData> {
+    let subject: Certificate = rasn::der::decode(&certs[0]).ok()?;
+    let issuer: Certificate = rasn::der::decode(&certs[1]).ok()?;
+
+    let issuer_name_raw = rasn::der::encode(&issuer.tbs_certificate.subject).ok()?;
+    let issuer_key_raw = &issuer
+        .tbs_certificate
+        .subject_public_key_info
+        .subject_public_key
+        .as_raw_slice();
+
+    let issuer_name_hash = OctetString::from(crate::crypto::hash::sha1(&issuer_name_raw));
+    let issuer_key_hash = OctetString::from(crate::crypto::hash::sha1(issuer_key_raw));
+    let serial_number = subject.tbs_certificate.serial_number;
+
+    // Build request structures
+    let sha1_oid = rasn::types::Oid::new(&[1, 3, 14, 3, 2, 26])?;
+    let alg = rasn::types::ObjectIdentifier::from(sha1_oid);
+
+    let sha1_ai = rasn_pkix::AlgorithmIdentifier {
+        algorithm: alg,
+        parameters: Some(Any::new(rasn::der::encode(&()).ok()?)),
+        // Many OCSP responders expect this to be NULL not None.
+    };
+
+    let req_cert = rasn_ocsp::CertId {
+        hash_algorithm: sha1_ai,
+        issuer_name_hash,
+        issuer_key_hash,
+        serial_number,
+    };
+
+    let ocsp_req = rasn_ocsp::Request {
+        req_cert,
+        single_request_extensions: None,
+    };
+
+    let request_list = vec![ocsp_req];
+
+    let tbs_request = rasn_ocsp::TbsRequest {
+        version: rasn_ocsp::Version::from(0u8),
+        requestor_name: None,
+        request_list,
+        request_extensions: None,
+    };
+
+    let ocsp_request = rasn_ocsp::OcspRequest {
+        tbs_request,
+        optional_signature: None,
+    };
+
+    let request_der = rasn::der::encode(&ocsp_request).ok()?;
+    let request_str = base64::encode(&request_der);
+    let url = url::Url::parse(responder_url).ok()?;
+
+    Some(OcspRequestData { request_str, url })
+}
+
+// Common function to process OCSP responders
+#[cfg(any(not(target_arch = "wasm32"), target_os = "wasi"))]
+fn process_ocsp_responders<F>(certs: &[Vec<u8>], mut fetch_fn: F) -> Option<Vec<u8>>
+where
+    F: FnMut(&OcspRequestData) -> Option<Vec<u8>>,
+{
```

Comment:
Maybe this function should just return a list of ocsp requests? It doesn't seem right to have it only for sync code yet inline the same code for async. It also doesn't seem right to have this function support both async/sync.